core/sail/memory/src/test/java/org/eclipse/rdf4j/sail/memory/QueryPlanRetrievalTest.java

```diff
@@ -87,32 +87,32 @@ public void testTupleQuery() {
 			String actual = query.explain(Explanation.Level.Unoptimized).toString();
 
 			String expected = "Projection\n" +
-					"   ProjectionElemList\n" +
-					"      ProjectionElem \"a\"\n" +
-					"   Filter\n" +
-					"      Compare (!=)\n" +
-					"         Var (name=c)\n" +
-					"         Var (name=d)\n" +
-					"      LeftJoin\n" +
-					"         Join\n" +
-					"            Join\n" +
-					"               LeftJoin (new scope)\n" +
-					"                  SingletonSet\n" +
-					"                  StatementPattern\n" +
-					"                     Var (name=d)\n" +
-					"                     Var (name=e)\n" +
-					"                     Var (name=f)\n" +
-					"               StatementPattern\n" +
-					"                  Var (name=a)\n" +
-					"                  Var (name=_const_f5e5585a_uri, value=http://www.w3.org/1999/02/22-rdf-syntax-ns#type, anonymous)\n"
+					"╠══ProjectionElemList\n" +
+					"║     ProjectionElem \"a\"\n" +
+					"╚══Filter\n" +
+					"   ├──Compare (!=)\n" +
+					"   │     Var (name=c)\n" +
+					"   │     Var (name=d)\n" +
+					"   └──LeftJoin\n" +
+					"      ╠══Join\n" +
+					"      ║  ├──Join\n" +
+					"      ║  │  ╠══LeftJoin (new scope)\n" +
```

Comment:
If I understood correctly, the idea is to use the double lines for all nodes that have two (or more) children, single lines only for unary predicates. So why is the outer join here single lined, and the inner one double?

Reply:
It just swaps every other node. So that it’s easier to follow the lines visually. Double vs single has no meaning.